emulator/src/kreator/assembler.rs

```diff
@@ -0,0 +1,101 @@
+use std::fmt;
+
+struct Assembler {
+    code: Vec<String>,
+    program_counter: u16,
+}
+
+impl Assembler {
+    fn new(input_code: &str) -> Assembler {
+        let mut lines: Vec<String> = Vec::new();
+        for line in input_code.split("\n") {
+            let line_parts = line.trim().split(";").collect::<Vec<&str>>();
```

Comment:
I think collect() can infer the type here, so the type annotation is unnecessary.

Reply:
collect() [can not infer the type](https://users.rust-lang.org/t/why-rust-can-not-infer-the-results-type-of-collect/30844/4)  making the type annotation a requirement. (Notice that _line_parts_ is not explicitly a Vector, unlike _lines_)

Reply:
Also, putting a type annotation on the variable, instead of specifying the generic type is more readable, e.g let lineparts: Vec<&str>  = ...collect()
atleast in my opinion

Reply:
another option would be to use strip_suffix (https://doc.rust-lang.org/std/string/struct.String.html#method.strip_suffix) which i just found with a clojure